virshPool.py

```diff
@@ -0,0 +1,50 @@
+import logging
+from typing import Optional
+
+import host
+from logger import logger
+
+
+class VirshPool:
```

Comment:
Could we use a data class here instead of a regular python class? Doing so will make this class look a lot cleaner.

Reply:
`VirshPool` indeed is stateless (it has no mutable fields and doesn't change after construction). I think a dataclass should also be immutable and simple, so `VirshPool` may qualify.

Still, the methods have some non-trivial logic that goes beyond what a dataclass IMO should do. I think dataclass can have complex methods, as long as they don't have side effects. E.g. to validate the class. But `VirshPool` seems to do too much.

Yes, it could also be a dataclass. I would prefer leave it however.

Reply:
@vrindle can you elaborate on your idea?

Reply:
At the end of the day it’s not a big deal either way. That said, using a dataclass could eliminate the constructor, all of the getters, and enforce immutable immutability with frozen=True

A dataclass is just a normal class with a little syntactic sugar to get rid of the tedious parts. It might be helpful here 😄

Reply:
ok, I changed my mind.

The result is clearly nice.

I also changed it, because https://peps.python.org/pep-0557/ makes it clear that a dataclass is just a regular class where several methods get autogenerated. So, it doesn't necessarily need to be only for holding data.